tests/src/FunctionalJavascript/Form/ImporterFormTest.php

```diff
@@ -0,0 +1,205 @@
+
+/**
+ * Tests adding and editing feeds migrate importers.
+ *
+ * @group feeds_migrate
+ */
+class ImporterFormTest extends FeedsMigrateJavascriptTestBase {
+
+  /**
+   * Tests adding a new importer.
+   */
+  public function testAddImporter() {
+    $migration = $this->createMigration();
+    $migration->set('source', [
+      'plugin' => 'url',
+      'data_fetcher_plugin' => 'http',
+      'data_parser_plugin' => 'simple_xml',
+      'item_selector' => '/items/item',
+    ]);
+    $migration->set('destination', [
+      'plugin' => 'entity:node',
+      'default_bundle' => 'article',
+    ]);
+    $migration->save();
+
+    // Navigate to create a new importer.
+    $this->drupalGet('/admin/content/feeds-migrate/importer/add');
+    $page = $this->getSession()->getPage();
+
+    // Set label and wait for machine name element to appear.
+    $label = $page->findField('edit-label');
+    $label->setValue('Importer A');
+    $this->assertSession()->waitForElementVisible('css', '#edit-label-machine-name-suffix .admin-link');
+
+    // Set import frequency to 'As often as possible'.
+    $import_frequency = $page->findField('edit-importfrequency');
+    $import_frequency->setValue(0);
+
+    // Set processor setting to 'Update existing content'.
+    $page->selectFieldOption('existing', 'update');
+
+    // Set processor setting to 'Keep orphaned items'.
+    $keep_orphans = $page->findField('edit-keeporphans');
+    $keep_orphans->check();
+
+    // Set migration id to 'Migration Simple XML'.
+    $migration_id = $page->findField('edit-migrationid');
+    $migration_id->selectOption($migration->id());
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Assert source configuration matches up with our migration.
+    $source_tab = $page->find('css', '[href="#plugin_settings--source"]');
+    $source_tab->click();
+    $source_plugin = $page->findField('migration[source][plugin]');
+    $this->assertTrue($migration->get('source')['plugin'], $source_plugin->getValue(), 'Source plugin matches with migration.');
+
+    // Assert data fetcher configuration matches up with our migration.
+    $data_fetcher_tab = $page->find('css', '[href="#plugin_settings--data_fetcher"]');
+    $data_fetcher_tab->click();
+    $data_fetcher_plugin = $page->findField('migration[source][data_fetcher_plugin]');
+    $this->assertTrue($migration->get('source')['data_fetcher_plugin'], $data_fetcher_plugin->getValue(), 'Data fetcher plugin matches with migration.');
+
+    // Assert data fetcher 'File Location' is visible.
+    $data_fetcher_urls = $this->assertSession()->fieldExists('source_wrapper[importer][data_fetcher_wrapper][importer][urls]');
+    // Field location should be empty.
+    $this->assertTrue(empty($data_fetcher_urls->getValue()), 'File location is empty.');
+    // Set the file location to an API endpoint.
+    $data_fetcher_urls->setValue('http://news.com/api/articles');
```

Comment:
While in the test no import is done from this url, I would avoid using real looking urls to prevent people from taking this over as example and have their tests depend on the external availability of a site. For test urls, always use www.example.com.